crypto/x509v3/pcy_cache.c

```diff
@@ -53,11 +54,17 @@ static int policy_cache_create(X509 *x,
         } else if (sk_X509_POLICY_DATA_find(cache->data, data) != -1) {
             ret = -1;
             goto bad_policy;
-        } else if (!sk_X509_POLICY_DATA_push(cache->data, data))
-            goto bad_policy;
+        } else if (!sk_X509_POLICY_DATA_push(cache->data, data)) {
+            goto memerr;
+        }
         data = NULL;
     }
     ret = 1;
+
+    if (0) {  /* always false */
```

Comment:
This construction looks awkward to me. And again: the error message should immediately follow the failed function.